txselector/txselector.go

```diff
@@ -699,7 +699,7 @@ func (txsel *TxSelector) processL2Txs(
 				l2Txs[i].Info = obj.Message
 				l2Txs[i].ErrorCode = obj.Code
 				l2Txs[i].ErrorType = obj.Type
-				nonSelectedL2Txs = append(nonSelectedL2Txs, l2Txs[i])
+				unforjableL2Txs = append(unforjableL2Txs, l2Txs[i])
```

Comment:
Could you explain this change?

The difference between `nonSelectedL2Txs` vs `unforjableL2Txs`:

- nonSelectedL2Txs: Txs that have been discarded in the current iteration, but could get selected in next iterations (example: another tx that is processed later on gives enough balance to the account so the tx can finally get forged)
- unforjableL2Txs: Txs that cannot be forged in this iterartion nor in future iterations (example: nonce for the account is already greater than the one signed in the tx)

I believe in this case it should be `nonSelectedL2Txs`

Reply:
This specific error means that the coordinator does not have the authorization needed in order to create an account on behalf of the receiver, we discussed this situation and decided that for this specific scenario we should only try to forge this tx once per batch, if it's not possible, it will wait until all the iterations of this batch are over to be processed again in the next one. 

I know you @arnaubennassar know this, but just to make it clear to everyone reading this thread, even though it is marked as unforgeable, the tx still remains in the pool as a pending state for the next forge cycle, if the authorization is sent to the node in the mean time, this tx will be forged in the next batch.

Reply:
Very well explained @tclemos. Although I've read the function that returns the error (`txsel.processTxToEthAddrBJJ`), and I believe there is one case where it shouldn't be marked as unforgeable: https://github.com/hermeznetwork/hermez-node/blob/da2809b0d319235a89ce20e0d42c553e8f7c2afb/txselector/txselector.go#L964

Even if this case is very unlikely to happen, I think it's better to be safe in this circumstances

Reply:
This is fine because the reason to not forge at this time is the amount of txs being forged and not the lack of authorization to create the account.

Reply:
Exactly. The fact that at a given point the batch looks like full it's not enough to guarantee that the tx is not forgeable.

Think about the tons of optimizations we can do to the selector... some of them for sure will imply selecting and unselecting txs.

I don't want to block this PR more time, so if you think that it's better to leave it like this, it's good enough for me, the impact will most likely be meaningless.